msteams-platform/bots/how-to/conversations/bot-copilot-handoff.md

```diff
@@ -0,0 +1,178 @@
+
+   ```typescript
+    public async onInvokeActivity(context: TurnContext): Promise<InvokeResponse> {
+        try {
+          switch (context.activity.name) {
+            case "handoff/action": {
+              // TODO: Save continuation token and use it to process final response to user later
+             return {status: 200}; // return just the http status
+            }
+            case "composeExtension/query":
+              return {
+                status: 200,
+                body: await this.handleTeamsMessagingExtensionQuery(
+                  context,
+                  context.activity.value
+                ),
+              };
+            default:
+              return {
+                status: 200,
+                body: `Unknown invoke activity handled as default- ${context.activity.name}`,
+              };
+          }
+        } catch (err) {
+          console.log(`Error in onInvokeActivity: ${err}`);
+          return {
+            status: 500,
+            body: `Invoke activity received- ${context.activity.name}`,
+          };
+        }
+      }
+   ```
+
+   When the bot receives the invoke call, `context.activity.value.continuation` contains the `continuationToken` that was set in the deep link URL.
+
+## Best practices
+
+* It’s important for the bot to quickly notify users when they're directed to the bot chat after selecting the action button, as there's no visual indication. This helps manage expectations as there might be a delay before the bot returns a response due to network latency and processing time. For instance, the bot can send a series of activities to keep the user informed of the progress:
+
+    ```typescript
+    await context.sendActivities([
+      {
+        type: ActivityTypes.Message,
+        text: "Continuing conversation from Copilot...",
+      },
+      { type: ActivityTypes.Typing },
+      { type: "delay", value: 1000 },
+      {
+        type: ActivityTypes.Message,
+        text: `Fetching more details using the continuation token passed: ${continuationToken}`,
+      },
+      { type: ActivityTypes.Typing },
+      { type: "delay", value: 4000 },
+      {
+        type: ActivityTypes.Message,
+        text: `Handoff successful!`,
+      }
+    ]);
+    
+    ```
+
+* For a smooth handoff process, it’s important to manage the continuation token effectively. Keep the token’s lifetime short and ensure it doesn't process more than once. After a token is used, remove it from storage to prevent it from being used again. If the same token comes up, let the user know they need to start a new conversation with the bot because the handoff from Copilot can't continue.
+
+* When the bot is unable to process a request or if the token is expired, it must actively guide the user to the bot chat. If the conversation can't continue, the bot must inform the user of the error and suggest they start a new conversation or retry their query.
```

Comment:
Is the recommendation in this case different from the one in the previous bullet point? I'm reading and re-reading the two and I'm not sure I understand the difference.

Reply:
removed the second bullet point.